plugin/eureka/eureka_client.go

```
@@ -0,0 +1,66 @@
+package eureka
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	statusUp   status = "UP"
+	statusDown status = "DOWN"
+)
+
+type status string
+
+type serverResponse struct {
+	Applications *applications `json:"applications"`
+}
+type applications struct {
+	Application []*application `json:"application"`
+}
+type application struct {
+	Name     string      `json:"name"`
+	Instance []*instance `json:"instance"`
+}
+type instance struct {
+	IpAddr     string `json:"ipAddr"`
+	VipAddress string `json:"vipAddress"`
+	Status     status `json:"status"`
+}
+
+type clientAPI interface {
+	fetchAllApplications() (*applications, error)
+}
+
+type client struct {
+	clientAPI
+
+	BaseUrl string
+}
+
+func (e client) fetchAllApplications() (*applications, error) {
+	req, _ := http.NewRequest(http.MethodGet, e.BaseUrl+"/v2/apps/", nil)
+	req.Header.Set("Accept", "application/json")
+	client := &http.Client{}
+	resp, _ := client.Do(req)
```

Don't ignore this error.